pyne/alara.py

```diff
@@ -198,6 +199,112 @@ def photon_source_to_hdf5(filename, nucs='all', chunkshape=(10000,)):
+    quick later use.
+
+    This function produces a single HDF5 file named <filename>.h5 containing the
+    table headings:
+
+        idx : int
+            The volume element index assuming the volume elements appear in xyz
+            order (z changing fastest) within the photon source file in the case of
+            a structured mesh or mesh.mesh_iterate() order for an unstructured mesh.
+        nuc : str
+            The nuclide name as it appears in the output file.
+        time : str
+            The decay time as it appears in the output file.
+        response : float
+            Supported response:
+                - decay_heat [W/cm3]
+                - specific_activity [Bq/cm3]
+                - alpha_heat [W/cm3]
+                - beta_heat [W/cm3]
+                - gamma_heat [W/cm3]
+                - wdr
+                - photon_source
+
+    Parameters
+    ----------
+    filename : str
+        The path to the file output.txt
+    response : str
+        Key word of the response
+    chunkshape : tuple of int
+        A 1D tuple of the HDF5 chunkshape.
+    """
+    f = open(filename, 'r')
+    f.seek(0)
+
+    dt = np.dtype([
+        ('idx', np.int64),
+        ('nuc', 'S6'),
+        ('time', 'S20'),
+        (response, np.float64)])
+
+    filters = tb.Filters(complevel=1, complib='zlib')
+    h5_filename = os.path.join(os.path.dirname(filename), ''.join([response, '.h5']))
+    h5f = tb.open_file(h5_filename, 'w', filters=filters)
+    tab = h5f.create_table('/', 'data', dt, chunkshape=chunkshape)
+
+    chunksize = chunkshape[0]
+    rows = np.empty(chunksize, dtype=dt)
+    idx = 0
+    count = 1
+    decay_times = []
+    zone_start = False
+    response_start = False
+    for i, line in enumerate(f, 1):
+        # terminate condition
+        if str_elements_in_line('Totals for all zones.', line) \
+                and response_start:
+            break
+        # get response string
+        if zone_start and is_response_string(line, response):
+            response_start = True
+            continue
+        # get decay times
+        if str_elements_in_line('isotope shutdown', line):
+            if len(decay_times) == 0:
+                decay_times = read_decay_times(line)
+            continue
+        # get zone idx
+        if 'Zone #' in line:
+            idx = _get_zone_idx(line)
+            if idx == 0:
+                ## new blocks, disable first response string
+                #response_start = False
+                zone_start = True
+            continue
+        # skip the lines does not contain specific response info
+        if not response_start:
+            continue
+        # skip the lines don't contain wanted data
+        if not _is_data(line):
+            continue
+    
+        ls = line.strip().split()
+        # put data into table
+        # format of each row: idx, nuc, time, decay_heat
+        for k, dc in enumerate(decay_times):
+            j = (count-1) % chunksize
```

Comment:
This line is only used in one place, so it might be good to move it to just before that place.

Some might argue that this could just be done inline, without introducing this variable `j` at all, but it might help readability a little bit.  It might be better if it was called `chunk_row` or something?

Reply:
Updated this part in https://github.com/pyne/pyne/pull/1123/commits/e6069dce639cc742d7e8b30e4f65420e4c0d848b. Now `j` is defined just before it is used.